diff -uN -r stage07/www/index.html stage08/www/index.html
--- stage07/www/index.html	2025-10-19 16:59:28.926642667 +0900
+++ stage08/www/index.html	2025-10-19 16:59:28.543654779 +0900
@@ -8,6 +8,7 @@
     <script src="src/stage.js"></script>
     <script src="src/game.js"></script>
     <script src="src/image.js"></script>
+    <script src="src/player.js"></script>
 </head>
 
 <body style="margin: 0;">
diff -uN -r stage07/www/src/config.js stage08/www/src/config.js
--- stage07/www/src/config.js	2025-10-19 17:42:22.302732944 +0900
+++ stage08/www/src/config.js	2025-10-19 17:42:22.306732813 +0900
@@ -8,7 +8,6 @@
 
   // 初期状態のステージ
   static initialBoard = [
-    [0, 0, 0, 0, 4, 5],
     [0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0],
@@ -16,10 +15,11 @@
     [0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0],
-    [0, 1, 2, 3, 0, 0],
-    [1, 2, 3, 4, 5, 0],
-    [1, 2, 3, 4, 5, 0],
-    [1, 2, 3, 4, 5, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
   ];
 
   static puyoColorMax = 5; // 何色のぷよを使うか
@@ -27,4 +27,7 @@
   static erasePuyoCount = 4; // 何個以上揃ったら消えるか
   static eraseAnimationFrames = 30; // 何フレームでぷよを消すか
   static zenkeshiDuration = 150; // 全消し時のアニメーションミリセカンド
+
+  static playerFallingSpeed = 0.9; // プレイ中の自然落下のスピード
+  static playerLockDelayFrames = 20; // 何フレーム接地したらぷよを固定するか
 }
diff -uN -r stage07/www/src/game.js stage08/www/src/game.js
--- stage07/www/src/game.js	2025-10-19 17:42:22.303732911 +0900
+++ stage08/www/src/game.js	2025-10-19 17:42:22.307732781 +0900
@@ -58,7 +58,7 @@
           Stage.showZenkeshi();
         }
         comboCount = 0;
-        gameState = '';
+        gameState = 'createPlayerPuyo';
       }
       break;
     case 'erasingPuyo':
@@ -68,6 +68,27 @@
         gameState = 'checkFallingPuyo';
       }
       break;
+    case 'createPlayerPuyo':
+      // 新しくプレイヤーの操作ぷよを作成する状態
+      if (!Player.createPlayerPuyo()) {
+        // 新しい操作用ぷよを作成する。作成出来なかったら、ゲームオーバー
+        gameState = 'gameOver';
+      } else {
+        // プレイヤーが操作する
+        gameState = 'playing';
+      }
+      break;
+    case 'playing':
+      // プレイヤーが操作する状態
+      const nextAction = Player.update();
+      gameState = nextAction; // 'playing' 'fix' のどれかが帰ってくる
+      break;
+    case 'fix':
+      // 現在の位置でぷよを固定する状態
+      Player.fixPlayerPuyo();
+      // 固定が完了したら、自由落下出来るぷよがあるかどうかを確認する
+      gameState = 'checkFallingPuyo';
+      break;
   }
   frame++;
   setTimeout(gameLoop, 1000 / 60); // 1/60秒後にもう一度呼び出す
diff -uN -r stage07/www/src/player.js stage08/www/src/player.js
--- stage07/www/src/player.js	1970-01-01 09:00:00.000000000 +0900
+++ stage08/www/src/player.js	2025-10-19 17:42:22.308732749 +0900
@@ -0,0 +1,152 @@
+class Player {
+  static centerPuyoColor = 0;
+  static rotatingPuyoColor = 0;
+  static playerPuyoStatus = null;
+  static centerPuyoElement = null;
+  static rotatingPuyoElement = null;
+  static groundedFrame = 0;
+
+  // プレイヤーが操作するぷよを作る
+  static createPlayerPuyo() {
+    // ぷよぷよが置けるかどうか、1番上の段の左から3つ目を確認する
+    if (Stage.getPuyoInfo(2, 0)) {
+      // 空白でない場合は新しいぷよを置けない（ゲームオーバー）
+      return false;
+    }
+
+    // 新しいぷよの色を決める
+    Player.centerPuyoColor =
+      Math.trunc(Math.random() * Config.puyoColorMax) + 1;
+    Player.rotatingPuyoColor =
+      Math.trunc(Math.random() * Config.puyoColorMax) + 1;
+
+    // 新しいぷよ画像を作成する
+    Player.centerPuyoElement = GameImage.getPuyoImage(Player.centerPuyoColor);
+    Player.rotatingPuyoElement = GameImage.getPuyoImage(
+      Player.rotatingPuyoColor,
+    );
+    Stage.stageElement.appendChild(Player.centerPuyoElement);
+    Stage.stageElement.appendChild(Player.rotatingPuyoElement);
+
+    // ぷよの初期情報を定める
+    Player.playerPuyoStatus = {
+      x: 2, // 中心ぷよの位置: 左から2列目
+      y: -1, // 画面上部ギリギリから出てくる
+      left: 2 * Config.puyoImageWidth,
+      top: -1 * Config.puyoImageHeight,
+      dx: 0, // 中心ぷよから見た回転するぷよの相対位置:回転するぷよは上方向にある (0, -1)
+      dy: -1,
+      rotation: 90, // 画面上の回転するぷよの角度: 90度（上向き）
+    };
+    // 接地時間はゼロ
+    Player.groundedFrame = 0;
+    // ぷよを描画
+    Player.setPlayerPuyoPosition();
+    return true;
+  }
+
+  // playerPuyoStatus に従って、画面上のぷよの位置を更新する
+  static setPlayerPuyoPosition() {
+    // まず中心ぷよの位置を確定させる
+    Player.centerPuyoElement.style.left = Player.playerPuyoStatus.left + 'px';
+    Player.centerPuyoElement.style.top = Player.playerPuyoStatus.top + 'px';
+
+    // 次に回転するぷよの位置を計算する
+    const x =
+      Player.playerPuyoStatus.left +
+      Math.cos((Player.playerPuyoStatus.rotation * Math.PI) / 180) *
+        Config.puyoImageWidth;
+    const y =
+      Player.playerPuyoStatus.top -
+      Math.sin((Player.playerPuyoStatus.rotation * Math.PI) / 180) *
+        Config.puyoImageHeight;
+    Player.rotatingPuyoElement.style.left = x + 'px';
+    Player.rotatingPuyoElement.style.top = y + 'px';
+  }
+
+  // プレイヤーの操作ぷよを落下させる
+  static dropPlayerPuyo() {
+    let { x, y, dx, dy } = Player.playerPuyoStatus;
+
+    // 現状のプレイヤーの操作ぷよの下にぷよがあるか確認する
+    if (
+      !Stage.getPuyoInfo(x, y + 1) &&
+      !Stage.getPuyoInfo(x + dx, y + dy + 1)
+    ) {
+      // 中心ぷよ・回転するぷよ両方の下にぷよがないので、自由落下してよい
+      Player.playerPuyoStatus.top += Config.playerFallingSpeed;
+      // 自由落下した際、マス目の境界を超えていないか確認する
+      if (
+        Math.floor(Player.playerPuyoStatus.top / Config.puyoImageHeight) != y
+      ) {
+        // ブロックの境を超えたので、自分の位置をひとつ下にずらす
+        y += 1;
+        Player.playerPuyoStatus.y = y;
+        // 下にブロックがないかを再度チェックする
+        if (
+          !Stage.getPuyoInfo(x, y + 1) &&
+          !Stage.getPuyoInfo(x + dx, y + dy + 1)
+        ) {
+          // 境を超えたが、下にぷよはなかった。接地していないよう設定して、自由落下を続ける
+          Player.groundedFrame = 0;
+          return false;
+        } else {
+          // 境を超えたらぷよにぶつかった。位置を調節して、接地を開始する
+          Player.playerPuyoStatus.top = y * Config.puyoImageHeight;
+          Player.groundedFrame = 1;
+          return false;
+        }
+      } else {
+        // 自由落下でマス目を超えていなかった。接地していないよう設定して、自由落下を続ける
+        Player.groundedFrame = 0;
+        return false;
+      }
+    } else {
+      // プレイヤーの操作ぷよの下にぷよがあるので、必ず接地している
+      if (Player.groundedFrame === 0) {
+        // 初接地である。接地を開始する
+        Player.groundedFrame = 1;
+        return false;
+      } else {
+        // 接地中である。接地の時間を増加させる
+        Player.groundedFrame++;
+        if (Player.groundedFrame > Config.playerLockDelayFrames) {
+          // 接地で一定時間が経過した。この位置で固定する
+          return true;
+        } else {
+          return false;
+        }
+      }
+    }
+  }
+
+  // イベントループで現在の状況を更新する
+  static update() {
+    // まずプレイヤーの操作ぷよを落下させる
+    if (Player.dropPlayerPuyo()) {
+      // 接地が終わったら、ぷよを固定する
+      return 'fix';
+    }
+    // ぷよの位置を更新する
+    Player.setPlayerPuyoPosition();
+    return 'playing';
+  }
+
+  // 現在のプレイヤー操作ぷよを、ぷよぷよ盤の上に配置する
+  static fixPlayerPuyo() {
+    const { x, y, dx, dy } = Player.playerPuyoStatus;
+    if (y >= 0) {
+      // 中心ぷよが画面内にあった場合のみ配置する。画面外ならば配置しない
+      Stage.createPuyo(x, y, Player.centerPuyoColor);
+    }
+    if (y + dy >= 0) {
+      // 回転するぷよが画面内にあった場合のみ配置する。画面外ならば配置しない
+      Stage.createPuyo(x + dx, y + dy, Player.rotatingPuyoColor);
+    }
+    // 操作用に作成したプレイヤー操作ぷよ画像を画面上から消す
+    Player.centerPuyoElement.remove();
+    Player.centerPuyoElement = null;
+    Player.rotatingPuyoElement.remove();
+    Player.rotatingPuyoElement = null;
+  }
+}
