diff -u -r stage12/www/index.html stage13/www/index.html
--- stage12/www/index.html	2025-10-19 16:59:29.197634097 +0900
+++ stage13/www/index.html	2025-10-19 16:59:28.975641118 +0900
@@ -9,6 +9,7 @@
     <script src="src/game.js"></script>
     <script src="src/image.js"></script>
     <script src="src/player.js"></script>
+    <script src="src/score.js"></script>
 </head>
 
 <body style="margin: 0;">
@@ -16,6 +17,8 @@
     </div>
     <div id="stage" style="position:relative; margin: 0 auto; overflow: hidden; background: url(img/puyo_4bg.png)">
     </div>
+    <div id="score" style="position:relative; margin: 0 auto; overflow: hidden; text-align: right;">        
+    </div>
     <div style="display:none">
         <img src="img/puyo_1.png" id="puyo_1">
         <img src="img/puyo_2.png" id="puyo_2">
@@ -23,6 +26,16 @@
         <img src="img/puyo_4.png" id="puyo_4">
         <img src="img/puyo_5.png" id="puyo_5">
         <img src="img/zenkeshi.png" id="zenkeshi">
+        <img src="img/0.png" id="font_0">
+        <img src="img/1.png" id="font_1">
+        <img src="img/2.png" id="font_2">
+        <img src="img/3.png" id="font_3">
+        <img src="img/4.png" id="font_4">
+        <img src="img/5.png" id="font_5">
+        <img src="img/6.png" id="font_6">
+        <img src="img/7.png" id="font_7">
+        <img src="img/8.png" id="font_8">
+        <img src="img/9.png" id="font_9">
     </div>
 </body>
 
diff -u -r stage12/www/src/config.js stage13/www/src/config.js
--- stage12/www/src/config.js	2025-10-19 17:42:22.326732163 +0900
+++ stage13/www/src/config.js	2025-10-19 17:42:22.331732001 +0900
@@ -6,6 +6,7 @@
   static stageRows = 12; // ステージの縦の個数
   static stageBackgroundColor = '#11213b'; // ステージの背景色
   static nextBackgroundColor = '#e2a9c8'; // ネクスト表示の背景色
+  static scoreBackgroundColor = '#24c0bb'; // スコアの背景色
 
   // 初期状態のステージ
   static initialBoard = [
@@ -34,4 +35,13 @@
   static playerDownSpeed = 10; // プレイ中の下キー押下時の落下スピード
   static playerMoveFrames = 10; // 左右移動に消費するフレーム数
   static playerRotateFrames = 10; // 回転に消費するフレーム数
+
+  static scoreHeight = 33; // スコアのフォントの高さ
+  static comboBonusTable = [
+    0, 0, 8, 16, 32, 64, 96, 128, 160, 192, 224, 256, 288, 320, 352, 384, 416,
+    448, 480, 512, 544, 576, 608, 640, 672,
+  ]; // 連鎖ボーナステーブル
+  static pieceBonusTable = [0, 0, 0, 0, 0, 2, 3, 4, 5, 6, 7, 10]; // 連結ボーナステーブル
+  static colorBonusTable = [0, 0, 3, 6, 12, 24]; // 色数ボーナステーブル
+  static zenkeshiBonus = 3600; // 全消し時のボーナス
 }
diff -u -r stage12/www/src/game.js stage13/www/src/game.js
--- stage12/www/src/game.js	2025-10-19 17:42:22.327732131 +0900
+++ stage13/www/src/game.js	2025-10-19 17:42:22.332731968 +0900
@@ -21,6 +21,9 @@
   // プレイヤー操作を準備する
   Player.initialize();
 
+  // スコア表示を準備する
+  Score.initialize();
+
   // シーンを初期状態にセットする
   gameState = 'start';
   // フレームを初期化する
@@ -54,11 +57,15 @@
       if (eraseInfo) {
         gameState = 'erasingPuyo';
         comboCount++;
+        // スコアを加算する
+        Score.addComboScore(comboCount, eraseInfo.piece, eraseInfo.color);
         Stage.hideZenkeshi();
       } else {
         if (Stage.puyoCount === 0 && comboCount > 0) {
           // 全部消えたので、全消しを表示する
           Stage.showZenkeshi();
+          // ボーナススコアを加算する
+          Score.addScore(Config.zenkeshiBonus);
         }
         comboCount = 0;
         gameState = 'createPlayerPuyo';
diff -u -r stage12/www/src/image.js stage13/www/src/image.js
--- stage12/www/src/image.js	2025-10-19 17:42:22.327732131 +0900
+++ stage13/www/src/image.js	2025-10-19 17:42:22.333731935 +0900
@@ -1,5 +1,6 @@
 class GameImage {
   static puyoImageList = null;
+  static digitImageList = null;
 
   static initialize() {
     // ぷよ画像を準備する
@@ -12,6 +13,20 @@
       image.style.position = 'absolute';
       GameImage.puyoImageList[i] = image;
     }
+
+    // スコアの数字画像を準備する
+    GameImage.digitImageList = [];
+    for (let i = 0; i < 10; i++) {
+      const image = document.getElementById('font_' + i);
+      image.removeAttribute('id');
+
+      // 数字画像の高さをあらかじめ揃えておく
+      const width = (image.width / image.height) * Config.scoreHeight;
+      image.width = width;
+      image.height = Config.scoreHeight;
+
+      GameImage.digitImageList[i] = image;
+    }
   }
 
   // ぷよ画像を複製して返す
@@ -19,4 +34,15 @@
     const image = GameImage.puyoImageList[color - 1].cloneNode(true);
     return image;
   }
+
+  // 数字画像を複製して返す
+  static getDigitImage(digit) {
+    const image = GameImage.digitImageList[digit].cloneNode(true);
+    return image;
+  }
+
+  // 数字画像の横幅を返す
+  static getDigitImageWidth() {
+    return GameImage.digitImageList[0].width;
+  }
 }
diff -u -r stage12/www/src/player.js stage13/www/src/player.js
--- stage12/www/src/player.js	2025-10-19 17:42:22.329732065 +0900
+++ stage13/www/src/player.js	2025-10-19 17:42:22.336731838 +0900
@@ -144,6 +144,10 @@
         // ブロックの境を超えたので、自分の位置をひとつ下にずらす
         y += 1;
         Player.playerPuyoStatus.y = y;
+        // 下キーが押されていた場合、スコアを加算する
+        if (Player.keyStatus.down) {
+          Score.addScore(1);
+        }
         // 下にブロックがないかを再度チェックする
         if (
           !Stage.getPuyoInfo(x, y + 1) &&
Only in stage13/www/src: score.js
