diff -u -r stage06/www/index.html stage07/www/index.html
--- stage06/www/index.html	2025-10-19 16:59:28.714649371 +0900
+++ stage07/www/index.html	2025-10-19 16:59:28.926642667 +0900
@@ -19,6 +19,7 @@
         <img src="img/puyo_3.png" id="puyo_3">
         <img src="img/puyo_4.png" id="puyo_4">
         <img src="img/puyo_5.png" id="puyo_5">
+        <img src="img/zenkeshi.png" id="zenkeshi">
     </div>
 </body>
 
diff -u -r stage06/www/src/config.js stage07/www/src/config.js
--- stage06/www/src/config.js	2025-10-19 17:42:22.298733074 +0900
+++ stage07/www/src/config.js	2025-10-19 17:42:22.302732944 +0900
@@ -8,22 +8,23 @@
 
   // 初期状態のステージ
   static initialBoard = [
-    [0, 1, 2, 3, 4, 5],
-    [1, 2, 3, 4, 5, 0],
-    [2, 3, 4, 5, 0, 1],
+    [0, 0, 0, 0, 4, 5],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0],
-    [0, 1, 1, 1, 2, 0],
-    [0, 4, 5, 5, 2, 0],
-    [0, 4, 5, 5, 2, 0],
-    [0, 4, 3, 3, 3, 0],
-    [0, 0, 0, 0, 0, 0],
-    [1, 1, 0, 0, 2, 2],
-    [0, 0, 2, 2, 0, 0],
-    [4, 4, 0, 0, 5, 5],
+    [0, 0, 0, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0],
+    [0, 1, 2, 3, 0, 0],
+    [1, 2, 3, 4, 5, 0],
+    [1, 2, 3, 4, 5, 0],
+    [1, 2, 3, 4, 5, 0],
   ];
 
   static puyoColorMax = 5; // 何色のぷよを使うか
   static fallingSpeed = 6; // 自由落下のスピード
   static erasePuyoCount = 4; // 何個以上揃ったら消えるか
   static eraseAnimationFrames = 30; // 何フレームでぷよを消すか
+  static zenkeshiDuration = 150; // 全消し時のアニメーションミリセカンド
 }
diff -u -r stage06/www/src/game.js stage07/www/src/game.js
--- stage06/www/src/game.js	2025-10-19 17:42:22.299733041 +0900
+++ stage07/www/src/game.js	2025-10-19 17:42:22.303732911 +0900
@@ -51,7 +51,12 @@
       if (eraseInfo) {
         gameState = 'erasingPuyo';
         comboCount++;
+        Stage.hideZenkeshi();
       } else {
+        if (Stage.puyoCount === 0 && comboCount > 0) {
+          // 全部消えたので、全消しを表示する
+          Stage.showZenkeshi();
+        }
         comboCount = 0;
         gameState = '';
       }
diff -u -r stage06/www/src/stage.js stage07/www/src/stage.js
--- stage06/www/src/stage.js	2025-10-19 17:42:22.301732976 +0900
+++ stage07/www/src/stage.js	2025-10-19 17:42:22.308732749 +0900
@@ -5,6 +5,7 @@
   static fallingPuyoInfoList = [];
   static erasingStartFrame = 0;
   static erasingInfoList = [];
+  static zenkeshiImage = null;
 
   static initialize() {
     // HTML からステージの元となる要素を取得し、大きさを設定する
@@ -15,6 +16,13 @@
       Config.puyoImageHeight * Config.stageRows + 'px';
     Stage.stageElement.style.backgroundColor = Config.stageBackgroundColor;
 
+    // 全消しの画像を用意する
+    Stage.zenkeshiImage = document.getElementById('zenkeshi');
+    Stage.zenkeshiImage.width = Config.puyoImageWidth * Config.stageCols;
+    Stage.zenkeshiImage.style.position = 'absolute';
+    Stage.zenkeshiImage.style.opacity = '0';
+    Stage.stageElement.appendChild(Stage.zenkeshiImage);
+
     // ぷよぷよ盤を初期化する
     Stage.puyoCount = 0;
     Stage.puyoBoard = [];
@@ -271,4 +279,25 @@
       return true;
     }
   }
+
+  // 全消しの表示を開始する
+  static showZenkeshi() {
+    Stage.zenkeshiImage.style.transition = 'none';
+    Stage.zenkeshiImage.style.opacity = '1';
+    Stage.zenkeshiImage.style.top =
+      Config.puyoImageHeight * Config.stageRows + 'px';
+    Stage.zenkeshiImage.offsetHeight; // 一旦ここで表示を確定させる
+
+    Stage.zenkeshiImage.style.transition =
+      'top ' + Config.zenkeshiDuration + 'ms linear';
+    Stage.zenkeshiImage.style.top =
+      (Config.puyoImageHeight * Config.stageRows) / 3 + 'px';
+  }
+
+  // 全消しの画像を画面上から消す
+  static hideZenkeshi() {
+    Stage.zenkeshiImage.style.transition =
+      'opacity ' + Config.zenkeshiDuration + 'ms linear';
+    Stage.zenkeshiImage.style.opacity = '0';
+  }
 }
